libafl_qemu/src/snapshot.rs

```diff
@@ -151,12 +155,12 @@ impl QemuSnapshotHelper {
                 if map.flags().is_r() {
                     // TODO not just for R pages
                     unsafe {
-                        info.data = Some(Box::new(core::mem::zeroed()));
+                        info.data = Some(Box::new(Vec::with_capacity(self.page_size)));
```

Comment:
This will fail later: 'Vec::with_capacity` only sets the internal capacity of a vec but doesn't change the `len` of vec, so the line below `emulator.write_mem(*page, &data[..])` will get a 0-sized slice.
Either we want to change`write_mem` to take a `Vec` (which would mean we we don't have to 0-initialize the whole page) or use `vec![0; self.page_size];` here